lib/class-wp-rest-menu-custom-items-controller.php

```diff
@@ -0,0 +1,128 @@
+<?php
```

Comment:
Why can't think be part of the existing menu items controller?

Also why does this have to part of the REST API at all. Can't be loaded in via localize script or similar?

Reply:
The results are shown in the search results within `<LinkControl>` are retrieved on demand from the REST API [using this API within Gutenberg](https://github.com/WordPress/gutenberg/blob/f3ad70f2929c4092e355cfd30afd7f1059d6bc08/packages/core-data/src/fetch/__experimental-fetch-link-suggestions.js#L70).

It feels like the custom items should also be retrieved using the same mechanism. 

From the discussion in the PR thread it looks like there are a number of options being proposed so this may change direction.